cloud/auth/auth.go

```diff
@@ -289,25 +290,46 @@ func switchToLastUsedWorkspace(c *config.Context, workspaces []astro.Workspace)
 	return astro.Workspace{}, false, nil
 }
 
-// checkToken requests a users rolebindings and sets the workspace to make sure that token works
-// TODO check orgID is in the token
-func checkToken(c *config.Context, client astro.Client, out io.Writer) error {
-	self, err := client.GetUserInfo()
+// check client status after a successfully login
+func checkUserSession(c *config.Context, client astro.Client, coreClient astrocore.CoreClient, out io.Writer) error {
+	// fetch self user base on token
+	// we set CreateIfNotExist to true so we always create astro user when a successfully login
+	createIfNotExist := true
+	selfResp, err := coreClient.GetSelfUserWithResponse(http_context.Background(), &astrocore.GetSelfUserParams{
```

Comment:
why is it using `http_context` versus  `c *config.Context` which is being passed in?

Reply:
they are different thing right?  i think `c *config.Context` is our context from config file.

Reply:
you are correct.